### ISSUE TITLE
add support for env-file to the docker module

### ISSUE THREAD

**Issue by kampka** [ansible/ansible-modules-core#792]
Docker run supports the use of env-files for passing environment variables to containers.

```
  --env-file=[]              Read in a line delimited file of environment variables
```

It would be convenient if the docker module would support this as well.

**Comment by mahnunchik** [comment]
+1

Any news?

**Comment by MrMMorris** [comment]
+100 on this. It's a lot easier to use conditionals in a template than it is in a task if I want to omit some env vars

**Comment by vpetersson** [comment]
+1

**Comment by vpetersson** [comment]
@mpdehaan I'm inclined to write a fix for this, however before I do spend the time on that, can you tell me if you will accept this or if this is something that is already in the pipeline?

**Comment by bcoca** [comment]
@vpetersson I cannot find a PR in queue for this functionality

**Comment by vpetersson** [comment]
Ok, I'll try to carve out some time this weekend to write a patch then.

**Comment by vpetersson** [comment]
I took a closer look at this, and it appears as this is blocked by the lack of env-file support in docker-py.

I've gone ahead and opened an [issue](https://github.com/docker/docker-py/issues/565) over at the docker-py repo.

**Comment by dexterbt1** [comment]
+1 to this

**Comment by liquidvapour** [comment]
+1 We need this too. Now that it is in docker-py is that all that was needed to start work on getting it into ansible?

@vpetersson are you still up for doing this?

It would really clean up our build pipeline ;)

Thanks

**Comment by vpetersson** [comment]
@liquidvapour The code has been accepted and merged into docker-py and will be in the next version (I presume). I'm not really sure how closely Ansible tracks these dependencies, but we'd at the very least have to wait until the next build of docker-py is out.

**Comment by liquidvapour** [comment]
@vpetersson ok. Well I guess it is just a waiting game then ;) seems like you are all over it though. 

Looking forward to it.

**Comment by vpetersson** [comment]
Added an experimental version in PR #2155. Not sure if this is the right approach, but would love to get some feedback on it.

**Comment by denschu** [comment]
Any news here? Would love to see that feature in Ansible :+1:

**Comment by gesellix** [comment]
@denschu I'm going to check the PR #2155 tonight /cc @vpetersson

**Comment by gesellix** [comment]
@denschu see https://github.com/ansible/ansible-modules-core/pull/2155#issuecomment-177606142 for the current progress.

**Comment by denschu** [comment]
@gesellix Great, thanks for the fast feedback :-) Can you estimate how long it takes to get it into an official release?

**Comment by gesellix** [comment]
@denschu I have no idea or experience about the process for PRs in this project. Maybe @gregdek or @bcoca can tell us how to handle the PR as efficient as possible?

**Comment by cliffrowley** [comment]
Looking forward to this being released :-)

**Comment by thetrav** [comment]
+1

**Comment by ansibot** [comment]
@cove, @softzilla, @zfil, @ThomasSteinbach, ping. This issue is still waiting on your response.
[click here for bot help](https://github.com/jctanner/ansibullbot/blob/ISSUE_TRIAGE_CLASS/ISSUE_HELP.md)

**Comment by ansibot** [comment]
@cove, @softzilla, @zfil, @ThomasSteinbach, ping. This issue is still waiting on your response.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
@kampka We appreciate you taking the time to file this issue, but the module you've opened the issue about has been deprecated.

We will consider critical bugfixes for deprecated modules, but all other types of requests will be closed.

[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by horalstvo** [comment]
I'm missing exactly this feature in docker_service.
